src/backend/distributed/transaction/transaction_management.c

```diff
@@ -328,12 +328,15 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 				CoordinatedRemoteTransactionsAbort();
 			}
 
```

Comment:
I wonder if we can add a regression test that triggers below `ereport` call and so triggers the segfault on master:

```c
MultiConnection *
StartNodeUserDatabaseConnection(uint32 flags, const char *hostname, int32 port,
								const char *user, const char *database)
...........................................................................................................................
...........................................................................................................................
...........................................................................................................................
	/* do some minimal input checks */
	if (strlen(hostname) > MAX_NODE_LENGTH)
	{
		ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
						errmsg("hostname exceeds the maximum length of %d",
							   MAX_NODE_LENGTH)));
```

Reply:
(maybe by manually modifying `pg_dist_node` ?)

Reply:
I will have a look. I was thinking our catalog enforced the hostname length to <= 60 by using the `name` type. but it seems not to be the case.

Reply:
Interesting thought, will see if this is actually a plausible thing todo or that it will crash earlier.

Reply:
the nodename is actually truncated somewhere earlier:
```
[local] nilsdijk@postgres:9700-37881=# UPDATE pg_dist_node SET nodename = nodename || '-' || repeat('a',256-length(nodename));
UPDATE 2

[local] nilsdijk@postgres:9700-37881=# SELECT length(nodename) from pg_dist_node;
┌────────┐
│ length │
├────────┤
│    257 │
│    257 │
└────────┘
(2 rows)

[local] nilsdijk@postgres:9700-37881=# SELECT * FROM ref WHERE a = 1;
^CCancel request sent
WARNING:  connection to the remote node localhost-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa:9701 failed with the following error: could not translate host name "localhost-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" to address: nodename nor servname provided, or not known
ERROR:  connection to the remote node localhost-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa:9702 failed with the following error: could not translate host name "localhost-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" to address: nodename nor servname provided, or not known
Time: 139429.555 ms (02:19.430)
```

When debugging manually I see that the length of `hostname` at that point is 255, which is 2 characters less than what we store in the catalog table.